include/circt/Dialect/ESI/ESIPasses.td

```diff
@@ -0,0 +1,18 @@
+//===-- ESIPasses.td - ESI pass definition file ------------*- tablegen -*-===//
+//
+// This file contains definitions for passes that work on the ESI dialect.
+//
+//===----------------------------------------------------------------------===//
+
+#ifndef CIRCT_DIALECT_ESI_ESIPASSES_TD
+#define CIRCT_DIALECT_ESI_ESIPASSES_TD
+
+include "mlir/Pass/PassBase.td"
+
+def LowerESIToPhysical: Pass<"lower-esi-to-physical", "mlir::ModuleOp"> {
+  let summary = "Lower ESI abstract Ops to ESI physical ops.";
+  let constructor = "circt::esi::createESILoweringPass()";
+  let dependentDialects = ["circt::esi::ESIDialect"];
```

Comment:
I think in general, you don't need to depend on dialects that are part of the input. I think it's more for dialects that are not present in the input but may be generated in the output. In other words, I think this can be blank for a pass that lowers within the ESI.

Reply:
I just assumed that this was required since the ESI dialect isn't part of the ODS pass specification. I don't really understand what the dependent dialects are used for though. It does compile and work without it...

Reply:
It is mentioned in this FAQ: https://mlir.llvm.org/getting_started/Faq/#registered-loaded-dependent-whats-up-with-dialects-management

This is the relevant part:

> This is achieved by requiring every pass to declare a list of dependent Dialects: these are Dialects for which an entity (Operation, Type, or Attribute) can be created by the pass, other than for Dialects that would already be in the input.